decidim-core/app/helpers/decidim/decidim_form_helper.rb

```diff
@@ -25,7 +25,7 @@ def decidim_form_for(record, options = {}, &)
       if record.is_a?(ActiveRecord::Base)
         object = record.is_a?(Array) ? record.last : record
         format = options[:format]
-        apply_form_for_options!(record, object, options) if object
+        apply_form_for_options!(object, options) if object
```

Comment:
Change in the signature between Rails 6 and 7 
https://apidock.com/rails/v6.1.7.7/ActionView/Helpers/FormHelper/apply_form_for_options%21
https://apidock.com/rails/v7.0.0/ActionView/Helpers/FormHelper/apply_form_for_options%21